### PR TITLE
487 array layout

### LINKED ISSUE #487: Improve array control
From https://github.com/openmsupply/open-msupply/issues/464

Image

Think we also need a confirmation to remove an item from an array.

### PR DESCRIPTION
**Pull request by clemens-msupply**
- Better array support for string enum arrays
- Ask before removing item from an array

Closes #487

### CODE REVIEW

**Review by mark-prins — COMMENTED** [review]
Some initial comments... I'm having trouble testing the rest

**Comment by mark-prins** [comment]
Sorry! I don't like the usability of the string array:

image

My concerns:
- even though there is a thing selected, the `Select` component is blank
- the relation between the selecting of items and the display is unclear - having the action and result separated physically makes is hard to relate the two
- The list of selected items isn't clearly a list, just looks like text
- The hidden remove button is neat, but hard to use. It is unclear that you can remove items without accidentally hovering

I would recommend looking at using an Autocomplete with multiple options and the display of chips for selected elements:
https://mui.com/material-ui/react-autocomplete/#multiple-values

**Comment by mark-prins** [comment]
🤔 Is the warning alert really necessary? Personally, I think not. It is easy enough to add an item back in, so there is little cost in accidental removal, but an annoyance when removing.

File: client/packages/common/src/intl/locales/en/common.json [hunk 1]
```diff
@@ -217,6 +217,7 @@
   "log.requisition-status-finalised": "Requisition finalised",
   "messages.cant-delete-generic": "You cannot delete one or more of the selected items",
   "messages.confirm-delete-generic": "This will permanently remove data",
+  "messages.confirm-remove-item": "Are you sure to remove item?",
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply client/packages/common/src/intl/locales/en/common.json:220]: I use "Are you sure?" now

**Comment by clemens-msupply** [comment]
Thanks Autocomplete looks good:
image

Could do with some more spacing / padding (?)

**Review by mark-prins — APPROVED** [review]
😮  that is so good! fast work at getting the autocomplete done - very slick

**Comment by clemens-msupply** [comment]
thanks I did some slight refactoring for `findIndexOfRemoved` to make it hopefully clearer and also added some more checks for the input lists...

**Pull request merged**